Makefile

```
@@ -41,18 +30,20 @@ endif
 # Image URL to use all building/pushing image targets
 IMAGE ?= ${IMAGE_TAG_BASE}:${VERSION}
 
-OCI_BUILD_OPTS ?=
-
 # Image building tool (docker / podman) - docker is preferred in CI
 OCI_BIN_PATH = $(shell which docker 2>/dev/null || which podman)
 OCI_BIN ?= $(shell basename ${OCI_BIN_PATH})
+OCI_BUILD_OPTS ?=
+
+ifneq ($(CLEAN_BUILD),)
+	BUILD_DATE := $(shell date +%Y-%m-%d\ %H:%M)
+	BUILD_SHA := $(shell git rev-parse --short HEAD)
+	LDFLAGS ?= -X 'main.buildVersion=${VERSION}-${BUILD_SHA}' -X 'main.buildDate=${BUILD_DATE}'
+endif
```

In the end, only hours / minutes in the build date are forcing docker to regenerate the whole image.

Why not having just the date for local builds and full date + time on clean builds ?

```suggestion
OCI_BUILD_OPTS ?=
BUILD_DATE := $(shell date +%Y-%m-%d\)
ifneq ($(CLEAN_BUILD),)
	BUILD_DATE = $(shell date +%Y-%m-%d\ %H:%M)
endif
BUILD_SHA := $(shell git rev-parse --short HEAD)
BUILD_VERSION := $(TAG:v%=%)
ifneq ($(COMMIT), $(TAG_COMMIT))
	BUILD_VERSION := $(BUILD_VERSION)-$(BUILD_SHA)
endif
ifneq ($(shell git status --porcelain),)
	BUILD_VERSION := $(BUILD_VERSION)-dirty
endif
LDFLAGS := -X 'main.buildVersion=${BUILD_VERSION}' -X 'main.buildDate=${BUILD_DATE}'
```

---

(at the end of the day, do we really need all this stuff for dev builds? my goal was to simplify all of that in dev builds, and keep that only in CI builds)

---

For me it's useful for releases so I'm fine skipping these if you want to
These are only displayed in the query summary and logs.

FYI these info were coming from FLP Makefile https://github.com/netobserv/flowlogs-pipeline/blob/main/Makefile#L17-L28 so we may need to align other repos

---

yes I was planning to align other repos